erpnext/patches/v13_0/update_sla_enhancements.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2018, Frappe and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+
+import frappe
+
+def execute():
+	# add holiday list and employee group fields in SLA
+	# change response and resolution time in priorities child table
+	if frappe.db.exists('DocType', 'Service Level Agreement'):
+		sla_details = frappe.db.get_all('Service Level Agreement', fields=['name', 'service_level'])
+		priorities = frappe.db.get_all('Service Level Priority', fields=['*'], filters={
+			'parenttype': ('in', ['Service Level Agreement', 'Service Level'])
+		})
+
+		frappe.reload_doc('support', 'doctype', 'service_level_agreement')
+		frappe.reload_doc('support', 'doctype', 'service_level_priority')
+
+		for entry in sla_details:
+			values = frappe.db.get_value('Service Level', entry.service_level, ['holiday_list', 'employee_group'])
+			if values:
+				holiday_list = values[0]
+				employee_group = values[1]
+			frappe.db.set_value('Service Level Agreement', entry.name, {
+				'holiday_list': holiday_list,
+				'employee_group': employee_group
+			})
+
+		priority_dict = {}
+
+		for priority in priorities:
+			if priority.parenttype == 'Service Level Agreement':
```

Comment:
There are two things happening in the first part of the patch:

1. Updating the Service Level Priority child table in SLA DocType to change the response and resolution time. For that, I need to first have the old data with time and time_period from Service Level Priority DocType. Then I convert it to seconds as the fieldtype is duration.

2. Also, if there are no SLA documents using particular Service Level documents, I have created SLA from those so that data is not lost after deleting Service Level DocType. Such Service Levels are getting filtered [here](https://github.com/frappe/erpnext/pull/21617/files#diff-fcaf5cae6f782fad8791e921e526e83aR48).